### PR TITLE
refactor(go): updated go readmes

### PR DESCRIPTION
**Pull request by megglos**
## Description

Smaller scoped follow-up to https://github.com/camunda/zeebe/pull/15548 :
* updated commands in readmes
* regenerated stubs and mock for current state of protocol
* potential (but unlikely) flaki test fix

### CODE REVIEW

File: gateway-protocol-impl/README.md [hunk 1]
```diff
@@ -8,11 +8,11 @@ Zeebe client grpc protobuf protocol.
 
 To generate the go grpc stubs you need the following tools:
 
-1. Go >= 1.7 (https://golang.org/dl/)
+1. Go >= 1.19 (https://golang.org/dl/)
 2. The go protoc generator
 
    ```
-   go get -u github.com/golang/protobuf/protoc-gen-go
+   go install github.com/golang/protobuf/protoc-gen-go@v1.5.3
```
Comment (Zelldon) [inline gateway-protocol-impl/README.md:15]: 🔧 In general makes sense to have this version set here, but not sure whether this doesn't get out dated fast.

**Review by ChrisKujawa — APPROVED** [review]
Great. Thanks 👍🏼

File: gateway-protocol-impl/README.md (hunk 1 above)
Comment (megglos) [reply gateway-protocol-impl/README.md:15]: it's deprecated and no developed further :D the install command complained when no version was provided:
```
> go install github.com/golang/protobuf/protoc-gen-go
go: 'go install' requires a version when current directory is not in a module
```

Comment (Zelldon) [reply gateway-protocol-impl/README.md:15]: I guess latest would also work 🤔

Comment (megglos) [reply gateway-protocol-impl/README.md]: ah, TIL, adjusted :)

**Comment by megglos** [comment]
bors merge

**Pull request merged**